akka-stream/src/main/scala/akka/stream/impl/fusing/StreamOfStreams.scala

```diff
@@ -299,7 +299,7 @@ import akka.stream.impl.fusing.GraphStages.SingleSource
 
       private def tryCancel(): Boolean =
         // if there's no active substreams or there's only one but it's not been pushed yet
-        if (activeSubstreamsMap.isEmpty || (activeSubstreamsMap.size == substreamWaitingToBePushed.size)) {
+        if (activeSubstreamsMap.isEmpty || (activeSubstreamsMap.size == 1 && substreamWaitingToBePushed.isDefined)) {
```

Comment:
is this the same? it was previously `true` also for empty activeSubstreamsMap and None substreamWaitingToBePushed

Reply:
It is now also `true` for empty `activeSubstreamsMap` because that's the part before the `||` ;).

Took me a while to parse as well, but I think the new version is actually easier to read.